File: backend/src/Application/ReadingTracking/AchievementDefinitions.cs
namespace Application.ReadingTracking;

public record AchievementDef(string Code, string Category, string Name, string Description);

public static class AchievementDefinitions
{
    public static readonly Dictionary<string, AchievementDef> All = new()
    {
        // Session milestones
        ["first_session"] = new("first_session", "milestone", "First Steps", "Complete your first reading session"),
        ["first_book"] = new("first_book", "milestone", "Bookworm", "Finish your first book"),
        ["books_5"] = new("books_5", "milestone", "Avid Reader", "Finish 5 books"),
        ["books_10"] = new("books_10", "milestone", "Bibliophile", "Finish 10 books"),
        ["books_25"] = new("books_25", "milestone", "Scholar", "Finish 25 books"),
        ["books_50"] = new("books_50", "milestone", "Grandmaster", "Finish 50 books"),

        // Streak
        ["streak_3"] = new("streak_3", "streak", "Getting Started", "3-day reading streak"),
        ["streak_7"] = new("streak_7", "streak", "Week Warrior", "7-day reading streak"),
        ["streak_14"] = new("streak_14", "streak", "Fortnight Focus", "14-day reading streak"),
        ["streak_30"] = new("streak_30", "streak", "Monthly Master", "30-day reading streak"),
        ["streak_100"] = new("streak_100", "streak", "Century Club", "100-day reading streak"),

        // Time
        ["hours_1"] = new("hours_1", "time", "One Hour Down", "Read for 1 hour total"),
        ["hours_10"] = new("hours_10", "time", "Dedicated", "Read for 10 hours total"),
        ["hours_50"] = new("hours_50", "time", "Committed", "Read for 50 hours total"),
        ["hours_100"] = new("hours_100", "time", "Centurion", "Read for 100 hours total"),

        // Special
        ["early_bird"] = new("early_bird", "special", "Early Bird", "Read between 5-7 AM"),
        ["night_owl"] = new("night_owl", "special", "Night Owl", "Read between 11 PM - 3 AM"),
        ["speed_reader"] = new("speed_reader", "special", "Speed Reader", "Read over 400 words per minute"),
        ["marathon"] = new("marathon", "special", "Marathon", "Single session over 2 hours"),
    };
}

File: apps/web/src/lib/achievementDefinitions.ts
export interface AchievementDef {
  emoji: string
  name: string
  description: string
  category: string
}

export const AchievementDefinitions: Record<string, AchievementDef> = {
  // Milestones
  first_session: { emoji: '👣', name: 'First Steps', description: 'Complete your first reading session', category: 'milestone' },
  first_book: { emoji: '📖', name: 'Bookworm', description: 'Finish your first book', category: 'milestone' },
  books_5: { emoji: '📚', name: 'Avid Reader', description: 'Finish 5 books', category: 'milestone' },
  books_10: { emoji: '🏆', name: 'Bibliophile', description: 'Finish 10 books', category: 'milestone' },
  books_25: { emoji: '🎓', name: 'Scholar', description: 'Finish 25 books', category: 'milestone' },
  books_50: { emoji: '👑', name: 'Grandmaster', description: 'Finish 50 books', category: 'milestone' },

  // Streak
  streak_3: { emoji: '🔥', name: 'Getting Started', description: '3-day reading streak', category: 'streak' },
  streak_7: { emoji: '⚡', name: 'Week Warrior', description: '7-day reading streak', category: 'streak' },
  streak_14: { emoji: '💪', name: 'Fortnight Focus', description: '14-day reading streak', category: 'streak' },
  streak_30: { emoji: '🌟', name: 'Monthly Master', description: '30-day reading streak', category: 'streak' },
  streak_100: { emoji: '💎', name: 'Century Club', description: '100-day reading streak', category: 'streak' },

  // Time
  hours_1: { emoji: '⏱️', name: 'One Hour Down', description: 'Read for 1 hour total', category: 'time' },
  hours_10: { emoji: '⏰', name: 'Dedicated', description: 'Read for 10 hours total', category: 'time' },
  hours_50: { emoji: '🕰️', name: 'Committed', description: 'Read for 50 hours total', category: 'time' },
  hours_100: { emoji: '🏅', name: 'Centurion', description: 'Read for 100 hours total', category: 'time' },

  // Special
  early_bird: { emoji: '🌅', name: 'Early Bird', description: 'Read between 5-7 AM', category: 'special' },
  night_owl: { emoji: '🦉', name: 'Night Owl', description: 'Read between 11 PM - 3 AM', category: 'special' },
  speed_reader: { emoji: '⚡', name: 'Speed Reader', description: 'Read over 400 words per minute', category: 'special' },
  marathon: { emoji: '🏃', name: 'Marathon', description: 'Single session over 2 hours', category: 'special' },
}
